### DISCUSSION TITLE
Out of memory problem

### DISCUSSION THREAD

**Discussion by dpolivaev** [freeplane/freeplane#757] [Announcements]
Hello,

I have found one critical bug existing in Freeplane very long time.
If a mind map contains any presentation, it is saved at the root node. But if the root node with presentations is copied to any other node or any other mind map, the presentations are growing in the geometric progression each time when the map is saved.

So in the mind map submitted for analysis by @wordmuse he has got 262144 presentations.

Now I fixed the multiplication of the mind maps. But it is not unlikely that some users are facing this problem and are not aware of it. Therefore in addition to the bug fix (no preview available yet), I could do one of the following fixes:

- Check the content of all presentations and remove all duplicates
- Check the names of all presentations and remove all presentations with repeated names keeping only the first one
- Keep no more than 100 or 1000 presentations

Which solution would you choose and why?

**Comment by wordmuse** [comment]
I must have tried presentations at one time to see how it works, and
whether I could use it. I couldn't find a use case that worked for me for
the file that blew up on me. I don't have reason to present my mindmaps in
a dynamic way to anyone else. I can usually just take a screen capture of
the mindmap in one or two states, and that suffices.

So my answer to your 3 choices has to be taken with the above in mind.

I think that getting rid of duplicates would be good, and maybe at the same
time, if someone tries to create a presentation with a duplicate name, ask
if they want to overwrite the older presentation, or ask that they give the
new presentation a new name, thus avoiding duplication in the first place.

Hope this helps.

~ Bal

**Comment by macmarrum** [comment]
The usual workflow when creating presentations is to give a unique name to each one.
I know that Freeplane allows to have non-unique presentation names, but I believe it is quite rare for people to do that.
Taking this into consideration, I believe that is should be OK to check the names of all presentations and remove all presentations with repeated names keeping only the first one. It might make sense to make sure there's a backup before this action, so that User can restore it.
